docs/intent_conversation_api.md

```diff
@@ -0,0 +1,230 @@
+| `type` | string | Target type. One of `area`, `domain`, `device_class`, `device`, `entity`, or `custom`. |
+| `name` | string | Name of the affected target.                                                           |
+| `id`   | string | Optional. Id of the target.                                                            |
+
+An intent can have multiple targets which are applied on top of each other. The targets must be ordered from general to specific:
+
+* `area`
+  * A [registered area](https://developers.home-assistant.io/docs/area_registry_index/)
+* `domain`
+  * Home Assistant integration domain, such as "light"
+* `device_class`
+  * Device class for a domain, such as "garage_door" for the "cover" domain
+* `device`
+  * A [registered device](https://developers.home-assistant.io/docs/device_registry_index)
+* `entity`
+  * A [Home Assistant entity](https://developers.home-assistant.io/docs/architecture/devices-and-services)
+
+Most intents end up with 0, 1 or 2 targets. 3 targets currenly only happens when device classes are involved. Examples of target combinations:
+
+ * "Turn on the kitchen lights"
+     * 2 targets: `area:kitchen`, `domain:light`
+ * "Turn off all lights"
+     * 2 targets: `domain:light`, `entity:all`
+ * "Open the kitchen blinds"
+     * 3 targets: `area:kitchen`, `domain:cover`, `device_class:blind`
+
+
+### Query Answer
+
+The response is an answer to a question, such as "what is the temperature?". See the [speech](#speech) property for the answer text.
+
+```json
+{
+  "response": {
+    "response_type": "query_answer",
+    "language": "en",
+    "speech": {
+      "plain": {
+        "speech": "It is 65 degrees"
+      }
+    }
+  },
+  "conversation_id": "<generated-id-from-ha>",
+}
+```
+
+
+### Error
+
+An error occurred either during intent recognition or handling. See `data.code` for the specific type of error, and the [speech](#speech) property for the error message.
+
+```json
+{
+  "response": {
+    "response_type": "error",
+    "language": "en",
+    "data": {
+      "code": "no_intent_match"
+    },
+    "speech": {
+      "plain": {
+        "speech": "Sorry, I didn't understand that"
+      }
+    }
+  }
+}
+```
+
+`data.code` is a string that can be one of:
+
+* `no_intent_match` - The input text did not match any intents.
+* `no_valid_targets` - The targeted area, device, or entity does not exist.
+* `failed_to_handle` - An unexpected error occurred while handling the intent.
+* `unknown` - An error occurred outside the scope of intent processing.
+
+
+## Speech
+
+The spoken response to the user is provided in the `speech` property of the response. It can either be plain text (the default), or [SSML](https://www.w3.org/TR/speech-synthesis11/).
+
+For plain text speech, the response will look like:
+
+```json
+{
+  "response": {
+    "response_type": "...",
+    "speech": {
+      "plain": {
+        "speech": "...",
+        "extra_data": null
```

Comment:
if we don't have use for `extra_data` yet, should we just kill it ?

Reply:
Yeah, any scenario I can think of to use it should probably just be SSML anyways.